rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateGradleWrapper.java

```diff
@@ -111,6 +111,9 @@ public String getDescription() {
     @Override
     public Validated<Object> validate() {
         Validated<Object> validated = super.validate();
+        if (wrapperUri != null && (version != null || distribution != null)) {
+            return Validated.invalid("wrapperUri", wrapperUri, "WrapperUri cannot be used with other parameters");
```

Comment:
Perhaps we can extract the version from the wrapperUri here and validate it's an exact version

Reply:
I'm not sure that would benefit anything. If it's a custom uri, it wouldn't necessarily need to have a version in it at all and it's on the user to ensure that it is a valid distribution URL in the first place.

Reply:
I tend to agree with @shanman190  here. 
We have different distributions (other than bin and all), or at least use that part of the distribution url to set some differentiation between different flavours for our wrappers. 

I think if the users mentions a wrapperUri, he/she should be aware of the correct path of the wrapper and no replacements need to be done anymore.